reactor-core/src/main/java/reactor/core/publisher/Flux.java

```diff
@@ -7544,6 +7594,7 @@ public boolean test(O o) {
 	static final BooleanSupplier ALWAYS_BOOLEAN_SUPPLIER = () -> true;
 	@SuppressWarnings("rawtypes")
 	static final Function        HASHCODE_EXTRACTOR      = Object::hashCode;
+	static final BiPredicate     OBJECT_EQUAL            = Object::equals;
```

Comment:
this equals function change is orthogonal... shelve it and isolate it in another PR / commit on master